### DISCUSSION TITLE
Auto-approve MR when Renovate is part of the CODEOWNERS

### DISCUSSION THREAD

**Discussion by fad3t** [renovatebot/renovate#25905] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.63

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

Hello,

I run Renovate against a repository that has a protected branch (main) which requires CODEOWNERS approval.
The renovate configuration is the following:

"main" configuration:
```json
{
  "platform": "gitlab",
  // omitted for brevity
  "reviewersFromCodeOwners": true,
  "lockFileMaintenance": {
    "enabled": true,
    "automerge": true,
    "autoApprove": true
  }
}
```

and the config specific to the repository:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "platformAutomerge": true,
  "gitLabIgnoreApprovals": true,
  "autoApprove": true
}
```

Renovate runs with a group access token, and the corresponding user was added to the CODEOWNERS file, so it should have permissions to approve MRs.

Looking at the debug logs, I see Renovate attempts to merge the MR but it's getting an HTTP 405 back, because the MR isn't approved. Also I don't see any HTTP call to the `/approve` API endpoint, so it looks like Renovate isn't even trying to approve the MR.

I don't want to use `gitlabIgnoreApprovals: true` as I need to keep the approval restriction in place -- and I think it doesn't work with protected branches anyway, and using `automergeType: branch` isn't an option either as I want to keep track of dependancy change with a MR.

I would rather expect Renovate to call the `/approve` API endpoint, and then merge the MR.
Am I missing something, or is it simply not supported (yet?) by Renovate?

Thanks a lot,
Fred

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
Questions:
1. You say "I don't want to use gitlabIgnoreApprovals: true as I need to keep the approval restriction in place" but in your config you show you have it
2. Your config shows autoApprove=true but saying you don't see Renovate approving? Please note that approving only happens at MR _creation_ so if you're testing on existing MRs, it won't work

**Reply by fad3t** [reply]
1. Correct, I wanted to give it a try and see if this could become a temporary workaround but it didn't work. I forgot to remove it from my config.
2. Indeed, I've tried by closing an old MR (and changing its title) and letting Renovate create a new one; same result.

**Reply by rarkins** [reply, marked as answer]
If (1) didn't work then there seems to be a bigger problem in your setup. i.e. if Renovate cannot approve when you configure it on all the time, it's also not going to be able to approve selectively per your idea. You need to fix (1) first

**Reply by fad3t** [reply]
Well I'm not sure, because the way approval rules work in GitLab is different from protected branch w/ CODEOWNERS. When using CODEOWNERS on a protected branch, it doesn't really appear as an approval rule, explaining why Renovate cannot ignore it. I think I've seen other discussions of people reporting the same behavior.

Is Renovate supposed to try to approve a MR where CODEOWNERS approval is required? Am I supposed to see this logged somewhere?

**Reply by fad3t** [reply]
Ow okay, I've just ran another test and it got properly approved as expected.
So I guess I mixed up some configs during my tests.

Removing the `gitlabIgnoreApprovals` does the job.

Thx for your help!
